libs/vkd3d-shader/spirv.c

```diff
@@ -5091,6 +5091,33 @@ static const struct vkd3d_shader_global_binding *vkd3d_dxbc_compiler_get_global_
             vkd3d_spirv_enable_capability(builder, SpvCapabilitySampledImageArrayNonUniformIndexingEXT);
         }
     }
+    else if (data_type == VKD3D_DATA_UAV)
```

Comment:
We're missing decorations here such as: Writeonly/Readonly and explicit format for non-typed UAVs. I think we need a unique variable for each variant of these combinations based on what the shader module wants to declare.